### PR TITLE
fix(include/outputs): Whitelist logic reformulated to exactly match input

### PR DESCRIPTION
**Pull request by n4ch04**
### Context 

prowler-cloud/prowler#961

### Description

Logic of whitelisting matching reformulated to exactly macth input

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/outputs [hunk 1]
```diff
@@ -169,14 +169,17 @@ textFail(){
   colorcode="$BAD"
   while read -r i; do
```
Comment (jfagoagas) [inline include/outputs]: What is this `i`?

I think a more expressive name would be better.

File: include/outputs [hunk 2]
```diff
@@ -169,14 +169,17 @@ textFail(){
...
     ignore_check_name="${i%%:*}" # Check name is everything up to the first :
-    ignore_value="${i#*${CHECK_NAME}:}" # Ignore value is everything after the first :
-    # Check to see if ignore value appears anywhere within log message.
-    resource_value=".*${ignore_value}.*"
+    # Resource value is the second field of line included in whitelist divided by :
+    resource_value=$(awk -F ":" '{print $2}' <<< $i)
+    # Checked value is the second field of log message divided by spaces
+    checked_value=$(awk '{print $2}' <<< $1)
+    echo 
     if [[ ${ignore_check_name} != "${CHECK_NAME}" ]]; then
```
Comment (jfagoagas) [inline include/outputs]:
```suggestion
    if [[ "${ignore_check_name}" != "${CHECK_NAME}" ]]; then
```

File: include/outputs [hunk 3]
```diff
@@ -169,14 +169,17 @@ textFail(){
...
       # not for this check
       continue
     fi
-    if [[ $1 =~ ${resource_value} ]]; then
+    # To set WARNING flag both values must be exactly the same
+    if [[ ${checked_value} == ${resource_value} ]]; then
```
Comment (jfagoagas) [inline include/outputs]:
```suggestion
    if [[ "${checked_value}" == "${resource_value}" ]]; then
```

File: include/outputs [hunk 4]
```diff
@@ -169,14 +169,17 @@ textFail(){
...
       level="WARNING"
       colorcode="$WARNING"
```
Comment (jfagoagas) [inline include/outputs]:
```suggestion
      colorcode="${WARNING}"
```

File: include/outputs [hunk 5]
```diff
@@ -169,14 +169,17 @@ textFail(){
...
   while read -r i; do
     ignore_check_name="${i%%:*}" # Check name is everything up to the first :
-    ignore_value="${i#*${CHECK_NAME}:}" # Ignore value is everything after the first :
-    # Check to see if ignore value appears anywhere within log message.
-    resource_value=".*${ignore_value}.*"
+    # Resource value is the second field of line included in whitelist divided by :
+    resource_value=$(awk -F ":" '{print $2}' <<< $i)
+    # Checked value is the second field of log message divided by spaces
+    checked_value=$(awk '{print $2}' <<< $1)
+    echo
```
Comment (jfagoagas) [inline include/outputs]: Is this `echo` necessary here?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: include/outputs (hunk 2 above)
Comment (n4ch04) [reply include/outputs]: Included

File: include/outputs (hunk 3 above)
Comment (n4ch04) [reply include/outputs]: Included

File: include/outputs (hunk 4 above)
Comment (n4ch04) [reply include/outputs]: Included

File: include/outputs (hunk 5 above)
Comment (n4ch04) [reply include/outputs]: Deleted

File: include/outputs (hunk 1 above)
Comment (n4ch04) [reply include/outputs]: changed, variable is called excluded_item instead

**Review by jfagoagas — APPROVED** [review]
Good job @n4ch04 !!

**Pull request merged**